lib/matplotlib/backends/backend_wx.py

```diff
@@ -1054,25 +1058,8 @@ def _icon(name):
         *name*, including the extension and relative to Matplotlib's "images"
         data directory.
         """
-        pilimg = PIL.Image.open(cbook._get_data_path("images", name))
-        # ensure RGBA as wx BitMap expects RGBA format
-        image = np.array(pilimg.convert("RGBA"))
-        try:
-            dark = wx.SystemSettings.GetAppearance().IsDark()
-        except AttributeError:  # wxpython < 4.1
-            # copied from wx's IsUsingDarkBackground / GetLuminance.
-            bg = wx.SystemSettings.GetColour(wx.SYS_COLOUR_WINDOW)
-            fg = wx.SystemSettings.GetColour(wx.SYS_COLOUR_WINDOWTEXT)
-            # See wx.Colour.GetLuminance.
-            bg_lum = (.299 * bg.red + .587 * bg.green + .114 * bg.blue) / 255
-            fg_lum = (.299 * fg.red + .587 * fg.green + .114 * fg.blue) / 255
-            dark = fg_lum - bg_lum > .2
-        if dark:
-            fg = wx.SystemSettings.GetColour(wx.SYS_COLOUR_WINDOWTEXT)
-            black_mask = (image[..., :3] == 0).all(axis=-1)
-            image[black_mask, :3] = (fg.Red(), fg.Green(), fg.Blue())
-        return wx.Bitmap.FromBufferRGBA(
-            image.shape[1], image.shape[0], image.tobytes())
```

Comment:
Doesn't this lose the compatibility with dark mode? I never could figure out how to test it, though.

Reply:
@QuLogic Thank you for your review. You are correct.

I switched from reading the icons from `png` format to `svg` format so they will appear sharp in all DPIs.

As you point out and I mentioned in the _additional details_, this format switch has the caveat that this "black theme" code doesn't work because PIL has no support for svg. Since I have never seen this code execute in practice, I decided the tradeoff to have sharp icons was acceptable. But others may have other opinions. 

Ideally, someone who uses and can test the "black theme" functionality in hidpi could suggest how to add it back while reading the icons in SVG format!

Reply:
From a very quick look, given that we generate the SVG icons ourselves, I think we can guarantee that their color can be switched with a simple string replacement, and thus it should be doable to generate the proper-colored SVG icons on the fly?

Reply:
Thank you @anntzer  I just looked into this!

Currently our monochromatic svg icons --which are the ones that need changing for black themes-- do not explicitly specify foreground color, so the foreground always appears black since that is the svg default. The background is explicitly specified as transparent which does not need changing for black themes.

I propose that we submit a separate PR making the foreground explicitly black in all monochromatic icons. This is straightforward. It simply requires adding `style="fill:black;"` in one spot of each svg file. I just tested this and it is straightforward.

If the foreground color is explicitly listed in the svg, then adding support for black theme in backends simply involves doing a string replacement `fill:black` -> `fill:white` as you suggest.

If you think that this is a reasonable path, I will submit a PR with the icon changes right away, and restore support for black themes in this PR assuming the changes in the other PR.

What do you think?

Reply:
Looks reasonable to me.

Reply:
@anntzer Thank you for your quick reply! 

Finished with the changes.

1. I just restored black theme support in the `wxAgg` backend in this PR.
1. I explicitly set the foreground color of svg icons to black in a separate PR: https://github.com/matplotlib/matplotlib/pull/26731

I tested both changes together and they work as expected.